packages/core/src/services/documents/update.test.ts

```diff
@@ -370,4 +370,48 @@ describe('updateDocument', () => {
       null,
     )
   })
+
+  it('creates a new document in the commit when dataset is passed', async (ctx) => {
+    const { workspace, project, user, documents } =
+      await ctx.factories.createProject({
+        providers: [
+          {
+            type: Providers.OpenAI,
+            name: 'openai',
+          },
+        ],
+        documents: {
+          doc1: ctx.factories.helpers.createPrompt({
+            provider: 'openai',
+            content: 'Doc 1',
+          }),
+        },
+      })
+    const docsScope = new DocumentVersionsRepository(project.workspaceId)
+
+    const { commit } = await ctx.factories.createDraft({ project, user })
+    const doc1 = documents.find((d) => d.path === 'doc1')!
+
+    const dataset = await ctx.factories
+      .createDataset({
+        name: 'Test Dataset',
+        workspace,
+        author: user,
+      })
+      .then((r) => r.dataset)
+    await updateDocument({
+      commit,
+      document: doc1,
+      dataset,
+    }).then((r) => r.unwrap())
+
+    await recomputeChanges({ draft: commit, workspaceId: workspace.id })
+
+    const commitDocs = await docsScope
+      .getDocumentsAtCommit(commit)
+      .then((r) => r.unwrap())
+
+    const newDoc = commitDocs.find((d) => d.path === 'doc1')
+    expect(newDoc?.datasetId).toBe(dataset.id)
```

Comment:
I don't think assigining a dataset to a document should trigger an update. I think it should even be doable in merged commits. This property does not "define" the document, just how it's used in the ui

Reply:
But what if I create a new draft that changes the document a lot? I could change the dataset used in that playground. 

Now production version of that document will be pointing to that dataset too

Reply:
yesyes I don't say the same dataset should apply to all versions. But I don't think changing a dataset should be treated as "a change to the document". If it does, I would not be able to change the dataset in a merged commit without having to create a new draft just for that.

Reply:
Talked offline. Carlos was right. Changed